runtime/gc_structs/ClassLoaderClassesIterator.cpp

```diff
@@ -99,6 +102,55 @@ GC_ClassLoaderClassesIterator::switchToSystemMode()
 	return isSystemClassLoader;
 }
 
+J9Class *
+GC_ClassLoaderClassesIterator::nextArrayClass()
+{
+	while (_arrayState != state_done) {
+		switch(_arrayState) {
+			case state_valueTypeArray:
+#if defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES)
+				if (NULL != J9CLASS_GET_NULLRESTRICTED_ARRAY(_startingClass)) {
+					_iterateArrayClazz = J9CLASS_GET_NULLRESTRICTED_ARRAY(_startingClass);
```

Comment:
I noticed `J9CLASS_GET_NULLRESTRICTED_ARRAY()` does check for `J9_IS_J9CLASS_VALUETYPE()`, so we are doing it twice. May be is make sense to call macro only once?
```
    J9Class *nullRestrictedArray = J9CLASS_GET_NULLRESTRICTED_ARRAY(_startingClass);
    if (NULL != nullRestrictedArray) {
        _iterateArrayClazz = nullRestrictedArray;
```

Reply:
I'm getting a compile error when I try this so the declaration of `J9Class *nullRestrictedArray` will go outside the for loop. Does that sound okay?

Reply:
Yes, you can move this line outside while loop, it is ok too. Regarding my original suggestion I think if you put `{}` around switch case body it should resolve compile problem.
```
			case state_valueTypeArray:
            {
#if defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES)
                J9Class *nullRestrictedArray = J9CLASS_GET_NULLRESTRICTED_ARRAY(_startingClass);
                if (NULL != nullRestrictedArray) {
.....
            }
            break;
```

Reply:
Okay I'll add the brackets instead.